content/en/blog/_posts/2023-12-18-kubernetes-1-29-feature-loadbalancer-ip-mode-alpha.md

```diff
@@ -0,0 +1,102 @@
+    Some cloud providers use the load balancer's IP as the source IP when 
+    transmitting packets to the node. In the ipvs mode of kube-proxy, 
+    there is a problem that health checks from the load balancer never return. This occurs because the reply packets 
+    would be forward to the local interface `kube-ipvs0`(where the load balancer's IP is bound to) 
+    and be subsequently ignored.
+  
+2. **[Feature loss at load balancer level](https://github.com/kubernetes/kubernetes/issues/66607):**
+    Certain cloud providers offer features(such as TLS termination, proxy protocol, etc.) at the
+    load balancer level.
+    Bypassing the load balancer results in the loss of these features when the packet reaches the service
+    (leading to protocol errors).
+  
+
+Currently, there is a [workaround](https://github.com/kubernetes/kubernetes/issues/66607#issuecomment-474513060) that involves setting the `hostname` of the `service.status.loabalancer.ingress` 
+to bypass kube-proxy binding. 
+But this is just a makeshift solution.
+
+## Solution
+
+In summary, providing an option for cloud providers to disable the current behavior would be highly beneficial.
+
+To address this, Kubernetes v1.29 introduces a new (alpha) `service.status.loadbalancer.ingress.ipMode` 
+field for a Service.
+This field specifies how the load balancer IP behaves and can be specified only when 
+the `.status.loadBalancer.ingress.ip` field is also specified.
+
+Two values are possible for `.status.loadBalancer.ingress.ipMode`: `"VIP"` and `"Proxy"`.
+The default value is "VIP", meaning that traffic delivered to the node 
+with the destination set to the load balancer's IP and port will be redirected to the backend service by kube-proxy.
+This preserves the existing behavior of kube-proxy. 
+The "Proxy" value is intended to prevent kube-proxy from binding the Load Balancer IP to the node in both ipvs and iptables modes. 
+Consequently, traffic is sent directly to the load balancer and then forwarded to the destination node. 
+The destination setting for forwarded packets varies depending on how the cloud provider's load balancer delivers traffic:
+
+- If the traffic is delivered to the node then DNATed to the pod, the destination would be set to the node's IP and node port;
+- If the traffic is delivered directly to the pod, the destination would be set to the pod's IP and port.
+
+## Usage
+
+Here are the necessary steps to enable this feature:
+
+- Download the [latest Kubernetes project](https://kubernetes.io/releases/download/) (version `v1.29.0` or later).
+- Enable the feature gate with the command line flag `--feature-gates=LoadBalancerIPMode=true` 
+on kube-proxy, kube-apiserver, and cloud-controller-manager.
+- Set `ipMode` for services which type is set to `LoadBalancer` to the appropriate value. 
+This step is likely handled by the cloud-controller-manager during the `EnsureLoadBalancer` process.
```

Comment:
```suggestion
This step is likely handled by your chosen cloud-controller-manager during the `EnsureLoadBalancer` process.
```
(in-tree cloud controller integrations? Not a thing in v1.29)

Reply:
Yes, it should be user's chosen cloud-controller-manager which may be provided by cloud providers.